

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>DESeq2 with phyloseq &mdash; MCBL 1.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/cutome.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Illumina’s BaseMount" href="mcbl-basemount.html" />
    <link rel="prev" title="Fastq adapter removal and QC" href="mcbl-tutorials-AD-clean.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> MCBL
          

          
          </a>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">About The MCBL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="mcbl.html">MCIC Computational Biology Lab (MCBL)</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-application.html">MCBL membership</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-computers.html">MCBL servers and computing resources</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="mcbl-datadownloads-hudsonalpha.html">Downloading from hudsonalpha.org</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-datadownloads-basespace.html">Downloading from BaseSpace</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-tutorials-PF-clean.html">Filter a CASAVA-generated fastq file</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-tutorials-AD-clean.html">Fastq adapter removal and QC</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">DESeq2 with phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-basemount.html">Illumina’s BaseMount</a></li>
</ul>
<p class="caption"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="mcbl-gbs.html">GBS</a></li>
<li class="toctree-l1"><a class="reference internal" href="microbiome-analysis.html">A basic microbiome analysis</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">MCBL</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>DESeq2 with phyloseq</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/mcbl-tutorials-DESeq2-phyloseq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <span class="target" id="module-DESeq2-phyloseq"></span><div class="figure align-right">
<img alt="_images/Logo.png" src="_images/Logo.png" />
</div>
<div class="section" id="deseq2-with-phyloseq">
<h1>DESeq2 with phyloseq<a class="headerlink" href="#deseq2-with-phyloseq" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<dl class="field-list simple">
<dt class="field-odd">Required OS</dt>
<dd class="field-odd"><p>OS x or Linux.</p>
</dd>
<dt class="field-even">Software</dt>
<dd class="field-even"><p>R, phyloseq R library</p>
</dd>
<dt class="field-odd">Purpose</dt>
<dd class="field-odd"><p>This document provides instructions about how to find differentially abundant OTUs for microbiome data.</p>
</dd>
<dt class="field-even">More</dt>
<dd class="field-even"><p>Read more about phyloseq DEseq2 <a class="reference external" href="http://joey711.github.io/phyloseq-extensions/DESeq2.html">here</a> and <a class="reference external" href="https://joey711.github.io/phyloseq/">here</a>.</p>
</dd>
<dt class="field-odd">Author</dt>
<dd class="field-odd"><p>This document was created by <a class="reference external" href="mailto:wijeratne&#46;3&#37;&#52;&#48;osu&#46;edu">Saranga Wijeratne</a>.</p>
</dd>
</dl>
</div>
<div class="section" id="software-installation">
<h2>Software installation<a class="headerlink" href="#software-installation" title="Permalink to this headline">¶</a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you are runing this on MCBL <em>mcic-sel019-d</em>, please skip the installation.
The following command in R-Studio will load the software module to your environment.</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="nf">library</span><span class="p">(</span><span class="s">&quot;phyloseq&quot;</span><span class="p">);</span> <span class="nf">packageVersion</span><span class="p">(</span><span class="s">&quot;phyloseq&quot;</span><span class="p">)</span>
<span class="n">[1]</span> ‘<span class="m">1.16.2</span>’ <span class="c1"># version</span>
</pre></div>
</td></tr></table></div>
</div>
<p>Install the <code class="docutils literal notranslate"><span class="pre">phyloseq</span></code> package as follows:</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="nf">source</span><span class="p">(</span><span class="s">&#39;http://bioconductor.org/biocLite.R&#39;</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="nf">biocLite</span><span class="p">(</span><span class="s">&#39;phyloseq&#39;</span><span class="p">)</span>
</pre></div>
</td></tr></table></div>
</div>
<div class="section" id="import-data-with-phyloseq">
<h2>Import data with phyloseq<a class="headerlink" href="#import-data-with-phyloseq" title="Permalink to this headline">¶</a></h2>
<p>For this step, you need Biom and mapping files generated by the Qiime pipeline.</p>
<dl class="field-list simple">
<dt class="field-odd">Input biom file</dt>
<dd class="field-odd"><p><cite>otu_table_mc10_w_tax.biom</cite></p>
</dd>
<dt class="field-even">Qiime mapping file</dt>
<dd class="field-even"><p><cite>mapping.txt</cite></p>
</dd>
<dt class="field-odd">Output file</dt>
<dd class="field-odd"><p><cite>DESeq2_Out</cite></p>
</dd>
</dl>
<p>Copy all the input files to your “Working Directory” before you execute the following commands.</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="c1"># Filenames:</span>
<span class="n">biom_file</span> <span class="o">&lt;-</span> <span class="s">&quot;otu_table_mc10_w_tax.biom&quot;</span>
<span class="n">mapping_file</span> <span class="o">&lt;-</span> <span class="s">&quot;mapping.txt&quot;</span>

<span class="c1"># Import the biom file with phyloseq:</span>
<span class="n">biom_otu_tax</span> <span class="o">&lt;-</span> <span class="nf">import_biom</span><span class="p">(</span><span class="n">biom_file</span><span class="p">)</span>

<span class="c1"># Import the mapping file with phyloseq:</span>
<span class="n">mapping_file</span> <span class="o">&lt;-</span> <span class="nf">import_qiime_sample_data</span><span class="p">(</span><span class="n">mapping_file</span><span class="p">)</span>

<span class="c1"># Merge biom and mapping files with phyloseq:</span>
<span class="n">merged_mapping_biom</span> <span class="o">&lt;-</span> <span class="nf">merge_phyloseq</span><span class="p">(</span><span class="n">biom_otu_tax</span><span class="p">,</span><span class="n">mapping_file</span><span class="p">)</span>

<span class="c1"># Set column names in the taxa table:</span>
<span class="nf">colnames</span><span class="p">(</span><span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;kingdom&quot;</span><span class="p">,</span> <span class="s">&quot;Phylum&quot;</span><span class="p">,</span> <span class="s">&quot;Class&quot;</span><span class="p">,</span> <span class="s">&quot;Order&quot;</span><span class="p">,</span> <span class="s">&quot;Family&quot;</span><span class="p">,</span> <span class="s">&quot;Genus&quot;</span><span class="p">,</span> <span class="s">&quot;species&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table></div>
<p>Now, your merged mapping and Biom output should look as follows:</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="n">merged_mapping_biom</span>

<span class="c1"># phyloseq-class experiment-level object</span>
<span class="c1"># otu_table()   OTU Table:         [ 315 taxa and 9 samples ]</span>
<span class="c1"># sample_data() Sample Data:       [ 9 samples by 8 sample variables ]</span>
<span class="c1"># tax_table()   Taxonomy Table:    [ 315 taxa by 7 taxonomic ranks ]</span>
</pre></div>
</td></tr></table></div>
<p>The mapping file should look like this:</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8
9</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="nf">head</span><span class="p">(</span><span class="n">mapping_file</span><span class="p">)</span>

<span class="c1"># Sample Data:        [40 samples by 7 sample variables]:</span>
<span class="c1"># X.SampleID BarcodeSequence LinkerPrimerSequence InputFileName IncubationDate Treatment Description</span>
<span class="c1"># S1          S1              NA                   NA      S1.fasta              0        CO         CO1</span>
<span class="c1"># S2          S2              NA                   NA      S2.fasta              0        CO         CO2</span>
<span class="c1"># S3          S3              NA                   NA      S3.fasta              0        CO         CO3</span>
<span class="c1"># S4          S4              NA                   NA      S4.fasta             15        CO         CO4</span>
<span class="c1"># S5          S5              NA                   NA      S5.fasta             15        CO         CO5</span>
</pre></div>
</td></tr></table></div>
<p>To remove taxonomy level tags assigned to each level (<a href="#id3"><span class="problematic" id="id4">k__</span></a>, <a href="#id3"><span class="problematic" id="id5">p__</span></a>, etc..), issue the following commands:</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;k__([[:alpha:]])&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span> <span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;p__([[:alpha:]])&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;c__([[:alpha:]])&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;o__([[:alpha:]])&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;f__([[:alpha:]])&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;g__([[:alpha:]])&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;s__([[:alpha:]])&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;p__(\\[)&quot;</span><span class="p">,</span><span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;c__(\\[)&quot;</span><span class="p">,</span><span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;o__(\\[)&quot;</span><span class="p">,</span><span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;f__(\\[)&quot;</span><span class="p">,</span><span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;g__(\\[)&quot;</span><span class="p">,</span><span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
<span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;s__(\\[)&quot;</span><span class="p">,</span><span class="s">&quot;\\1&quot;</span><span class="p">,</span> <span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">))</span>
</pre></div>
</td></tr></table></div>
</div>
<div class="section" id="testing-for-differential-abundance-among-otus">
<h2>Testing for differential abundance among OTUs<a class="headerlink" href="#testing-for-differential-abundance-among-otus" title="Permalink to this headline">¶</a></h2>
<dl class="field-list simple">
<dt class="field-odd">Input file</dt>
<dd class="field-odd"><p><cite>merged_mapping_biom</cite></p>
</dd>
<dt class="field-even">Output files</dt>
<dd class="field-even"><p><cite>DESeq2_Out.txt</cite></p>
</dd>
</dl>
<p><strong>1. Load the DESeq2 package into your R environment</strong></p>
<blockquote>
<div><div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;DESeq2&quot;</span><span class="p">)</span>
<span class="nf">packageVersion</span><span class="p">(</span><span class="s">&quot;DESeq2&quot;</span><span class="p">)</span>
    <span class="c1"># [1] ‘1.12.4’</span>
</pre></div>
</td></tr></table></div>
</div></blockquote>
<p><strong>2. Assign DESeq2 output name and padj-cutoff</strong></p>
<blockquote>
<div><div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="n">filename_out</span> <span class="o">&lt;-</span> <span class="s">&quot;DESeq2_Out.txt&quot;</span>
<span class="n">alpha</span> <span class="o">&lt;-</span> <span class="m">0.01</span>
</pre></div>
</td></tr></table></div>
</div></blockquote>
<dl>
<dt><strong>3. Convert to DESeqDataSet format</strong></dt><dd><p>The <code class="docutils literal notranslate"><span class="pre">phyloseq_to_deseq2()</span></code> function converts the phyloseq-format microbiome data (i.e merged_mapping_biom)
to a <code class="docutils literal notranslate"><span class="pre">DESeqDataSet</span></code> with dispersion estimated, using the experimental design formula (i.e ~ Treatment):</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="n">diagdds</span> <span class="o">&lt;-</span> <span class="nf">phyloseq_to_deseq2</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">,</span> <span class="o">~</span> <span class="n">Treatment</span><span class="p">)</span>
</pre></div>
</td></tr></table></div>
</dd>
</dl>
<p><strong>4. Run DESeq</strong></p>
<blockquote>
<div><blockquote>
<div><div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="highlight"><pre><span></span> <span class="n">diagdds</span> <span class="o">&lt;-</span> <span class="nf">DESeq</span><span class="p">(</span><span class="n">diagdds</span><span class="p">,</span> <span class="n">test</span><span class="o">=</span><span class="s">&quot;Wald&quot;</span><span class="p">,</span> <span class="n">fitType</span><span class="o">=</span><span class="s">&quot;parametric&quot;</span><span class="p">)</span>

 <span class="c1">## estimating size factors</span>
 <span class="c1">## estimating dispersions</span>
 <span class="c1">## gene-wise dispersion estimates</span>
 <span class="c1">## mean-dispersion relationship</span>
 <span class="c1">## final dispersion estimates</span>
 <span class="c1">## fitting model and testing</span>
</pre></div>
</td></tr></table></div>
</div></blockquote>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are getting the following error:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">Error</span> <span class="n">in</span> <span class="nf">estimateSizeFactorsForMatrix</span><span class="p">(</span><span class="nf">counts</span><span class="p">(</span><span class="n">object</span><span class="p">),</span> <span class="n">locfunc</span><span class="p">,</span> <span class="n">geoMeans</span> <span class="o">=</span> <span class="n">geoMeans</span><span class="p">)</span> <span class="o">:</span> <span class="n">every</span> <span class="n">gene</span> <span class="n">contains</span> <span class="n">at</span> <span class="n">least</span> <span class="n">one</span> <span class="n">zero</span><span class="p">,</span> <span class="n">cannot</span> <span class="n">compute</span> <span class="n">log</span> <span class="n">geometric</span> <span class="n">means</span>
<span class="n">Calls</span><span class="o">:</span> <span class="n">estimateSizeFactors</span> <span class="kc">...</span> <span class="n">estimateSizeFactors</span> <span class="o">-&gt;</span> <span class="n">.local</span> <span class="o">-&gt;</span> <span class="n">estimateSizeFactorsForMatrix</span>
</pre></div>
</div>
<p>Then please execute the following code (see <a class="reference external" href="https://github.com/joey711/phyloseq/issues/387">here</a> for more info):</p>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="n">gm_mean</span> <span class="o">&lt;-</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">na.rm</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">){</span> <span class="nf">exp</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="nf">log</span><span class="p">(</span><span class="n">x[x</span> <span class="o">&gt;</span> <span class="m">0</span><span class="n">]</span><span class="p">),</span> <span class="n">na.rm</span><span class="o">=</span><span class="n">na.rm</span><span class="p">)</span> <span class="o">/</span> <span class="nf">length</span><span class="p">(</span><span class="n">x</span><span class="p">))}</span>
<span class="n">geoMeans</span> <span class="o">&lt;-</span> <span class="nf">apply</span><span class="p">(</span><span class="nf">counts</span><span class="p">(</span><span class="n">diagdds</span><span class="p">),</span> <span class="m">1</span><span class="p">,</span> <span class="n">gm_mean</span><span class="p">)</span>
<span class="n">diagdds</span> <span class="o">&lt;-</span> <span class="nf">estimateSizeFactors</span><span class="p">(</span><span class="n">diagdds</span><span class="p">,</span> <span class="n">geoMeans</span> <span class="o">=</span> <span class="n">geoMeans</span><span class="p">)</span>
<span class="n">diagdds</span> <span class="o">&lt;-</span> <span class="nf">DESeq</span><span class="p">(</span><span class="n">diagdds</span><span class="p">,</span> <span class="n">test</span><span class="o">=</span><span class="s">&quot;Wald&quot;</span><span class="p">,</span> <span class="n">fitType</span><span class="o">=</span><span class="s">&quot;parametric&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table></div>
</div>
</div></blockquote>
<dl>
<dt><strong>5. Process the results</strong></dt><dd><blockquote>
<div><p>The <code class="docutils literal notranslate"><span class="pre">results</span></code> function creates a table of results. Then the <code class="docutils literal notranslate"><span class="pre">res</span></code> table is filtered by <code class="docutils literal notranslate"><span class="pre">padj</span> <span class="pre">&lt;</span> <span class="pre">alpha</span></code>.</p>
</div></blockquote>
<div class="highlight-r notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="n">res</span> <span class="o">&lt;-</span> <span class="nf">results</span><span class="p">(</span><span class="n">diagdds</span><span class="p">,</span> <span class="n">cooksCutoff</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">sigtab</span> <span class="o">&lt;-</span> <span class="n">res</span><span class="nf">[which</span><span class="p">(</span><span class="n">res</span><span class="o">$</span><span class="n">padj</span> <span class="o">&lt;</span> <span class="n">alpha</span><span class="p">),</span> <span class="n">]</span>
<span class="n">sigtab</span> <span class="o">&lt;-</span> <span class="nf">cbind</span><span class="p">(</span><span class="nf">as</span><span class="p">(</span><span class="n">sigtab</span><span class="p">,</span> <span class="s">&quot;data.frame&quot;</span><span class="p">),</span> <span class="nf">as</span><span class="p">(</span><span class="nf">tax_table</span><span class="p">(</span><span class="n">merged_mapping_biom</span><span class="p">)</span><span class="nf">[rownames</span><span class="p">(</span><span class="n">sigtab</span><span class="p">),</span> <span class="n">]</span><span class="p">,</span> <span class="s">&quot;matrix&quot;</span><span class="p">))</span> <span class="c1"># Bind taxonomic info to final results table</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="n">sigtab</span><span class="p">,</span> <span class="nf">as.character</span><span class="p">(</span><span class="n">filename_out</span><span class="p">))</span> <span class="c1"># Write `sigtab` to file</span>
</pre></div>
</td></tr></table></div>
</dd>
</dl>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="mcbl-basemount.html" class="btn btn-neutral float-right" title="Illumina’s BaseMount" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="mcbl-tutorials-AD-clean.html" class="btn btn-neutral float-left" title="Fastq adapter removal and QC" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2016-2020, Saranga Wijeratne &amp; Jelmer Poelstra.
      <span class="lastupdated">
        Last updated on Jul 06, 2020.
      </span>

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>