

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Illumina’s BaseMount &mdash; MCBL 1.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/cutome.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Download from hudsonalpha.org" href="mcbl-datadownloads-hudsonalpha.html" />
    <link rel="prev" title="Share runs and projects in BaseSpace" href="mcbl-share-basespace.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> MCBL
          

          
          </a>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">About The MCBL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="mcbl.html">MCIC Computational Biology Lab (MCBL)</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-application.html">MCBL membership</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-computers.html">MCBL servers and computing resources</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="mcbl-datadownloads-basespace.html">Download from BaseSpace</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-share-basespace.html">Share runs and projects in BaseSpace</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Illumina’s BaseMount</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-datadownloads-hudsonalpha.html">Download from hudsonalpha.org</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-tutorials-PF-clean.html">Filter a CASAVA-generated fastq file</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-tutorials-AD-clean.html">Fastq adapter removal and QC</a></li>
<li class="toctree-l1"><a class="reference internal" href="mcbl-tutorials-DESeq2-phyloseq.html">DESeq2 with phyloseq</a></li>
</ul>
<p class="caption"><span class="caption-text">Pipelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="mcbl-gbs.html">GBS</a></li>
<li class="toctree-l1"><a class="reference internal" href="microbiome-analysis.html">A basic microbiome analysis</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">MCBL</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Illumina’s BaseMount</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/mcbl-basemount.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <span class="target" id="module-basemount"></span><div class="figure align-right">
<img alt="_images/Logo.png" src="_images/Logo.png" />
</div>
<div class="section" id="illumina-s-basemount">
<h1>Illumina’s BaseMount<a class="headerlink" href="#illumina-s-basemount" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>1. Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<div class="section" id="file-access-traditional-way">
<h3>1.1. File Access (Traditional Way)<a class="headerlink" href="#file-access-traditional-way" title="Permalink to this headline">¶</a></h3>
<div class="figure align-left">
<img alt="_images/basespace_web_1.png" src="_images/basespace_web_1.png" />
</div>
</div>
<div class="section" id="file-access-with-basemount">
<h3>1.2. File Access (with BaseMount)<a class="headerlink" href="#file-access-with-basemount" title="Permalink to this headline">¶</a></h3>
<div class="figure align-left">
<img alt="_images/basemount_intro.jpg" src="_images/basemount_intro.jpg" />
</div>
</div>
<div class="section" id="what-is-basemount">
<h3>1.4. What is BaseMount?<a class="headerlink" href="#what-is-basemount" title="Permalink to this headline">¶</a></h3>
<p><strong>BaseMount</strong> is a mechanisum that enables access to your <strong>basespace storage</strong> as a <strong>Linux file system</strong>.</p>
</div>
<div class="section" id="advantages-of-basemount">
<h3>1.5. Advantages of BaseMount?<a class="headerlink" href="#advantages-of-basemount" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p>Have access to your Projects, Runs, and App results as your local files.</p></li>
<li><p>Can run <strong>local apps</strong> on basespace data <strong>without downloading data to your local computer.</strong></p></li>
<li><p>Can save your local storage space.:</p>
<blockquote>
<div><p>“Although BaseMount does facilitate file download, we would recommend that since BaseMount allows convenient, fast, cached access to your BaseSpace metadata and files, you may find that many operations can be carried out without the need to download locally. Duringour testing, we have used BaseMount to grep through fastq files, extract blocks of reads from bam files and even use IGV on the bam files directly all without downloading files locally. This can be more convenient than including a download step and saves on the overheads of local storage.” -Illumina</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="official-page">
<h3>1.6. Official page<a class="headerlink" href="#official-page" title="Permalink to this headline">¶</a></h3>
<div style="margin-top:10px;">
  <iframe width="900" height="600" src="https://basemount.basespace.illumina.com" frameborder="0" allowfullscreen></iframe>
</div></div>
</div>
<div class="section" id="basemount-installation">
<h2>2. BaseMount Installation<a class="headerlink" href="#basemount-installation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="quick-install">
<h3>2.1. Quick Install<a class="headerlink" href="#quick-install" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>sudo bash -c <span class="s2">&quot;</span><span class="k">$(</span>curl -L https://basemount.basespace.illumina.com/install/<span class="k">)</span><span class="s2">&quot;</span>
</pre></div>
</td></tr></table></div>
</div>
<div class="section" id="manual-install">
<h3>2.2. Manual install<a class="headerlink" href="#manual-install" title="Permalink to this headline">¶</a></h3>
<p><strong>Supported Operating Systems:</strong> Ubuntu, Centos</p>
<hr class="docutils" />
<p>Ubuntu 14 &amp; 15</p>
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="highlight"><pre><span></span>wget https://bintray.com/artifact/download/basespace/BaseSpaceFS-DEB/bsfs_1.1.631-1_amd64.deb
wget https://bintray.com/artifact/download/basespace/BaseMount-DEB/basemount_0.1.2.463-20150714_amd64.deb
sudo dpkg -i --force-confmiss bsfs_1.1.631-1_amd64.deb
sudo dpkg -i basemount_0.1.2.463-20150714_amd64.deb
</pre></div>
</td></tr></table></div>
</div>
<div class="section" id="mounting-your-basespace-account">
<h3>2.3. Mounting Your BaseSpace Account<a class="headerlink" href="#mounting-your-basespace-account" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="highlight"><pre><span></span>basemount --config <span class="o">{</span>config_file_prefix<span class="o">}</span> <span class="o">{</span>mount-point folder<span class="o">}</span>
basemount --config user1 ~/BaseSpace_Mount
</pre></div>
</td></tr></table></div>
<p><strong>Example:</strong></p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="highlight"><pre><span></span>mkdir /export/NFS/Saranga/BaseSpace
mkdir /export/NFS/Maria/BaseSpace
basemount --config Maria /export/NFS/Maria/BaseSpace/
</pre></div>
</td></tr></table></div>
<div class="figure align-left">
<img alt="_images/basemount_first.png" src="_images/basemount_first.png" />
</div>
<p>Then open your internet browser,</p>
<div class="figure align-left">
<img alt="_images/basemount_sec.png" src="_images/basemount_sec.png" />
</div>
<p>After you click “Accept”,</p>
<div class="figure align-left">
<img alt="_images/basemount_third.png" src="_images/basemount_third.png" />
</div>
<div class="figure align-left">
<img alt="_images/basemount_fourth.png" src="_images/basemount_fourth.png" />
</div>
<p>To access the folder, type,</p>
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="nb">cd</span> /export/NFS/Maria/BaseSpace/
ls
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>Projects  Runs
</pre></div>
</div>
<p>To see the contents in the Projects,</p>
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>ls -lsh /export/NFS/Saranga/BaseSpace/Projects/HiSeq<span class="se">\ </span><span class="m">2500</span><span class="se">\ </span>-<span class="se">\ </span>v4<span class="se">\ </span>reagents<span class="se">\:\ </span>TruSeq<span class="se">\ </span>PCR<span class="se">\ </span>Free<span class="se">\ \(</span><span class="m">4</span><span class="se">\ </span>replicates<span class="se">\ </span>of<span class="se">\ </span>NA12877<span class="se">\)</span>/Samples/NA12877_*/Files/
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>/export/NFS/Saranga/BaseSpace/Projects/MiSeq v3: TruSeq Targeted RNA Expression (NFkB &amp; Cell Cycle: Human Brain-Liver-UHRR)/Samples/Brain10/Files/:
total 85M
85M -r--r--r-- 1 root root 85M Oct  5 14:09 Brain10_S22_L001_R1_001.fastq.gz

/export/NFS/Saranga/BaseSpace/Projects/MiSeq v3: TruSeq Targeted RNA Expression (NFkB &amp; Cell Cycle: Human Brain-Liver-UHRR)/Samples/Brain11/Files/:
total 62M
62M -r--r--r-- 1 root root 62M Oct  5 14:09 Brain11_S23_L001_R1_001.fastq.gz
</pre></div>
</div>
</div>
</div>
<div class="section" id="basic-analysis-on-fastq-files">
<h2>3. Basic analysis on fastq files<a class="headerlink" href="#basic-analysis-on-fastq-files" title="Permalink to this headline">¶</a></h2>
<p>You can get basic information about your Fastq files without having to download them.
Such as:</p>
<ul class="simple">
<li><p>View sequences inside Fastq files</p></li>
<li><p>Number of reads for each Fastq file</p></li>
<li><p>Basic statistics and read length distribution</p></li>
</ul>
<p><strong>Example: View your data</strong></p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>zcat /export/NFS/Saranga/BaseSpace/Projects/MiSeq<span class="se">\ </span>v3<span class="se">\:\ </span>TruSeq<span class="se">\ </span>Targeted<span class="se">\ </span>RNA<span class="se">\ </span>Expression<span class="se">\ \(</span>NFkB<span class="se">\ \&amp;\ </span>Cell<span class="se">\ </span>Cycle<span class="se">\:\ </span>Human<span class="se">\ </span>Brain-Liver-UHRR<span class="se">\)</span>/Samples/Brain1/Files/Brain1_S13_L001_R1_001.fastq.gz <span class="p">|</span> head -n <span class="m">4</span>
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>@M03438:48:000000000-AGGNU:1:1101:11792:1006 1:N:0:13
NTCAATCCCCAGCAGTGGAATAAGGCCTGTTGTTCCTTGCAGTGGATCCTG
+
#88ABFFGCFEEG&lt;FF&lt;FDFFEGGFGGFCFGFFGGEGGGGGGFGGFGGGGG
</pre></div>
</div>
<p><strong>Example: Count the  number of sequneces using native Linux commands</strong></p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>zcat  /export/NFS/Saranga/BaseSpace/Projects/MiSeq<span class="se">\ </span>v3<span class="se">\:\ </span>TruSeq<span class="se">\ </span>Targeted<span class="se">\ </span>RNA<span class="se">\ </span>Expression<span class="se">\ \(</span>NFkB<span class="se">\ \&amp;\ </span>Cell<span class="se">\ </span>Cycle<span class="se">\:\ </span>Human<span class="se">\ </span>Brain-Liver-UHRR<span class="se">\)</span>/Samples/Brain1/Files/Brain1_S13_L001_R1_001.fastq.gz <span class="p">|</span> grep -c <span class="s2">&quot;@M03438:&quot;</span>
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>838876
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">FILE SIZE</dt>
<dd class="field-odd"><p>85M</p>
</dd>
<dt class="field-even">TIME TAKEN</dt>
<dd class="field-even"><p>1.327s</p>
</dd>
</dl>
<p><strong>Example: Count the  number of sequneces using using</strong> <a class="reference external" href="http://ngsutils.org/modules/fastqutils/">fastqutils</a> .</p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>fastqutils names /export/NFS/Saranga/BaseSpace/Projects/MiSeq<span class="se">\ </span>v3<span class="se">\:\ </span>TruSeq<span class="se">\ </span>Targeted<span class="se">\ </span>RNA<span class="se">\ </span>Expression<span class="se">\ \(</span>NFkB<span class="se">\ \&amp;\ </span>Cell<span class="se">\ </span>Cycle<span class="se">\:\ </span>Human<span class="se">\ </span>Brain-Liver-UHRR<span class="se">\)</span>/Samples/Brain1/Files/Brain1_S13_L001_R1_001.fastq.gz <span class="p">|</span> wc -l
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>838876
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">FILE SIZE</dt>
<dd class="field-odd"><p>85M</p>
</dd>
<dt class="field-even">TIME TAKEN</dt>
<dd class="field-even"><p>23.00s</p>
</dd>
</dl>
<p><strong>Example: Get the Length distribution and other statistics</strong></p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>fastqutils stats /export/NFS/Saranga/BaseSpace/Projects/MiSeq<span class="se">\ </span>v3<span class="se">\:\ </span>TruSeq<span class="se">\ </span>Targeted<span class="se">\ </span>RNA<span class="se">\ </span>Expression<span class="se">\ \(</span>NFkB<span class="se">\ \&amp;\ </span>Cell<span class="se">\ </span>Cycle<span class="se">\:\ </span>Human<span class="se">\ </span>Brain-Liver-UHRR<span class="se">\)</span>/Samples/Brain1/Files/Brain1_S13_L001_R1_001.fastq.gz
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>Space:   basespace
Pairing: Fragmented
Quality scale: Illumina
Number of reads:  838876

Length distribution
Mean: 51.0
StdDev:  0.0
Min:  51
25 percentile: 51
Median:  51
75 percentile: 51
Max:  51
Total:   838876

Quality distribution
pos   mean  stdev min   25pct 50pct 75pct max   count
1  33.5948650337  2.15033983948  2  34 34 34 34 838876
2  33.6285434319  2.01137931152  12 34 34 34 34 838876
3  33.6910246568  1.81306178651  11 34 34 34 34 838876
4  33.6861812711  1.84720681841  11 34 34 34 34 838876
5  33.6998292954  1.73881480815  12 34 34 34 34 838876
6  37.2788564698  2.49830369426  2  38 38 38 38 838876
7  37.3219820331  2.39576158974  2  38 38 38 38 838876
8  37.1795640834  2.72499009837  2  38 38 38 38 838876
9  37.158373824   2.76769991544  10 38 38 38 38 838876
10 37.0991517221  2.93041515545  10 38 38 38 38 838876
11 37.080357526   3.00286915879  10 38 38 38 38 838876
12 37.0752506926  2.93530617165  10 38 38 38 38 838876
13 37.1372705859  2.85104291311  10 38 38 38 38 838876
14 37.0559641711  3.00820427859  10 38 38 38 38 838876
15 37.0753877808  2.99577677236  10 38 38 38 38 838876
16 37.0950426523  2.92821324956  10 38 38 38 38 838876
17 37.204973083   2.64727500649  10 38 38 38 38 838876
18 37.1618308308  2.75627448135  10 38 38 38 38 838876
19 37.0932426247  2.92692822638  10 38 38 38 38 838876
20 37.1103548081  2.89001817524  10 38 38 38 38 838876
21 37.117659821   2.90476888945  10 38 38 38 38 838876
22 37.1280034236  2.80218109494  10 38 38 38 38 838876
23 36.9527546383  3.18334971719  9  37 38 38 38 838876
24 37.1825096915  2.70092644993  10 38 38 38 38 838876
25 37.2478948021  2.58021368225  10 38 38 38 38 838876
26 37.1056830807  3.04029966339  10 38 38 38 38 838876
27 37.0417129588  3.21755695865  9  38 38 38 38 838876
28 36.9245287742  3.46922882082  9  38 38 38 38 838876
29 37.0233538687  3.22132395112  9  38 38 38 38 838876
30 36.9768440151  3.36846192959  9  38 38 38 38 838876
31 36.9019378311  3.51733823479  10 38 38 38 38 838876
32 37.0442532627  3.15497307231  10 38 38 38 38 838876
33 37.0763462061  3.05842992907  10 38 38 38 38 838876
34 36.9112836701  3.51048896466  10 38 38 38 38 838876
35 36.871104907   3.50954115324  10 37 38 38 38 838876
36 36.9835911386  3.29022069717  9  38 38 38 38 838876
37 36.9526103977  3.40894509478  9  38 38 38 38 838876
38 36.9730198504  3.35198104312  10 38 38 38 38 838876
39 36.925962836   3.42925499742  9  38 38 38 38 838876
40 36.9641508399  3.3749237703   10 38 38 38 38 838876
41 36.9701290775  3.37108719426  9  38 38 38 38 838876
42 36.925310773   3.46541098262  9  38 38 38 38 838876
43 36.4323821399  4.37591782904  9  37 38 38 38 838876
44 36.6849510536  3.84804558398  10 37 38 38 38 838876
45 36.8000574578  3.71757395575  10 37 38 38 38 838876
46 36.743696327   3.89392229051  9  37 38 38 38 838876
47 36.6721195981  4.06597933483  10 37 38 38 38 838876
48 36.7998965282  3.73484635736  9  37 38 38 38 838876
49 36.9068074423  3.46537970313  10 38 38 38 38 838876
50 36.828634983   3.67970695764  10 37 38 38 38 838876
51 36.7344208202  3.75309659394  9  37 38 38 38 838876

Average quality string
BBBBBFFFFFFFFFFFFFFFFFEFFFFEFEEFFEEEEEEEEEEEEEEEEEE
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">FILE SIZE</dt>
<dd class="field-odd"><p>85M</p>
</dd>
<dt class="field-even">TIME TAKEN</dt>
<dd class="field-even"><p>1.10m</p>
</dd>
</dl>
</div>
<div class="section" id="basic-analysis-on-alignment-files-bam">
<h2>4. Basic analysis on Alignment files (BAM)<a class="headerlink" href="#basic-analysis-on-alignment-files-bam" title="Permalink to this headline">¶</a></h2>
<p><strong>Example: Check bam headers</strong></p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>samtools view -H /export/NFS/Saranga/BaseSpace/Projects/MiSeq<span class="se">\ </span>v3<span class="se">\:\ </span>TruSeq<span class="se">\ </span>Targeted<span class="se">\ </span>RNA<span class="se">\ </span>Expression<span class="se">\ \(</span>NFkB<span class="se">\ \&amp;\ </span>Cell<span class="se">\ </span>Cycle<span class="se">\:\ </span>Human<span class="se">\ </span>Brain-Liver-UHRR<span class="se">\)</span>/AppSessions/TopHat<span class="se">\ </span>Alignment<span class="se">\:\ </span>No<span class="se">\ </span>cSNP<span class="se">\ </span>or<span class="se">\ </span>RNA<span class="se">\ </span>Editing<span class="se">\ </span>found<span class="se">\ \(</span><span class="m">36</span><span class="se">\ </span>Samples<span class="se">\)</span>/AppResults.26970091.Brain1/Files/alignments/Brain1.alignments.bam
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>@HD   VN:1.0   SO:coordinate
@RG   ID:19 SM:Brain1
@SQ   SN:chrM  LN:16571
@SQ   SN:chr1  LN:249250621
@SQ   SN:chr2  LN:243199373
@SQ   SN:chr3  LN:198022430
@SQ   SN:chr4  LN:191154276
@SQ   SN:chr5  LN:180915260
@SQ   SN:chr6  LN:171115067
@SQ   SN:chr7  LN:159138663
@SQ   SN:chr8  LN:146364022
@SQ   SN:chr9  LN:141213431
@SQ   SN:chr10 LN:135534747
@SQ   SN:chr11 LN:135006516
@SQ   SN:chr12 LN:133851895
@SQ   SN:chr13 LN:115169878
@SQ   SN:chr14 LN:107349540
@SQ   SN:chr15 LN:102531392
@SQ   SN:chr16 LN:90354753
@SQ   SN:chr17 LN:81195210
@SQ   SN:chr18 LN:78077248
@SQ   SN:chr19 LN:59128983
@SQ   SN:chr20 LN:63025520
@SQ   SN:chr21 LN:48129895
@SQ   SN:chr22 LN:51304566
@SQ   SN:chrX  LN:155270560
@SQ   SN:chrY  LN:59373566
@PG   ID:TopHat   VN:2.0.7 CL:/illumina/development/IsisRNA/2.4.19.5/IsisRNA_Tools/bin/tophat --bowtie1 --read-realign-edit-dist 1 --segment-length 24 -o /data/input/Alignment/samples/Brain1/replicates/Brain1/tophat_main -p 1 --GTF /illumina/development/Genomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf --rg-id 19 --rg-sample Brain1 --library-type fr-firststrand --no-coverage-search --keep-fasta-order /illumina/development/Genomes/Homo_sapiens/UCSC/hg19/Sequence/BowtieIndex/genome /data/input/Alignment/samples/Brain1/replicates/Brain1/filtered/Brain1_S20_L001_R1_001.fastq.gz
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">FILE SIZE</dt>
<dd class="field-odd"><p>17M</p>
</dd>
<dt class="field-even">TIME TAKEN</dt>
<dd class="field-even"><p>0.006s</p>
</dd>
</dl>
<p><strong>Example: Check basic statistics on a Bam file</strong></p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>bamutils stats /export/NFS/Saranga/BaseSpace/Projects/MiSeq<span class="se">\ </span>v3<span class="se">\:\ </span>TruSeq<span class="se">\ </span>Targeted<span class="se">\ </span>RNA<span class="se">\ </span>Expression<span class="se">\ \(</span>NFkB<span class="se">\ \&amp;\ </span>Cell<span class="se">\ </span>Cycle<span class="se">\:\ </span>Human<span class="se">\ </span>Brain-Liver-UHRR<span class="se">\)</span>/AppSessions/TopHat<span class="se">\ </span>Alignment<span class="se">\:\ </span>No<span class="se">\ </span>cSNP<span class="se">\ </span>or<span class="se">\ </span>RNA<span class="se">\ </span>Editing<span class="se">\ </span>found<span class="se">\ \(</span><span class="m">36</span><span class="se">\ </span>Samples<span class="se">\)</span>/AppResults.26970091.Brain1/Files/alignments/Brain1.alignments.bam
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>Reads:   766531
Mapped:  766531
Unmapped:   0

Flag distribution
[0x010] Reverse complimented  383242   50.00%
[0x100] Secondary alignment   47 0.01%


Reference counts  count
chr1  32252
chr10 6829
chr11 45127
chr12 74524
chr13 24336
chr14 81664
chr15 254
chr16 5704
chr17 46662
chr18 9922
chr19 25644
chr2  32527
chr20 10708
chr21 22
chr22 7805
chr3  83585
chr4  42487
chr5  47865
chr6  106512
chr7  18024
chr8  6921
chr9  25334
chrM  0
chrX  31823
chrY  0
</pre></div>
</div>
<div class="toctree-wrapper compound">
</div>
</div>
<div class="section" id="metadata">
<h2>5. Metadata<a class="headerlink" href="#metadata" title="Permalink to this headline">¶</a></h2>
<p>In each directory, BaseMount provides a number of hidden files with extra BaseSpace metadata.
These are hidden files and their names start with a “.”.</p>
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="nb">cd</span> /export/NFS/Saranga/BaseSpace/Projects/MiSeq<span class="se">\ </span>v3<span class="se">\:\ </span>TruSeq<span class="se">\ </span>Targeted<span class="se">\ </span>RNA<span class="se">\ </span>Expression<span class="se">\ \(</span>NFkB<span class="se">\ \&amp;\ </span>Cell<span class="se">\ </span>Cycle<span class="se">\:\ </span>Human<span class="se">\ </span>Brain-Liver-UHRR<span class="se">\)</span>/Samples/
ls -l .?* <span class="c1">#List only the hidden files</span>
</pre></div>
</td></tr></table></div>
<div class="highlight-rest notranslate"><div class="highlight"><pre><span></span>-r-------- 1 swijeratne swijeratne        149 Nov 16 11:29 .curl
<span class="gh">.</span>
<span class="gh">.</span>
<span class="gh">.</span>
-r--r--r-- 1 swijeratne swijeratne      40674 Nov 16 11:29 .json
</pre></div>
</div>
<p><strong>Display content of the .json file</strong></p>
<hr class="docutils" />
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="highlight"><pre><span></span>cat .json
</pre></div>
</td></tr></table></div>
<p><strong>Query through a .json file qith “jq”</strong></p>
<div class="highlight-bash notranslate"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="highlight"><pre><span></span>cat .json <span class="p">|</span> jq <span class="s1">&#39;.Response.Items[] | select(.NumReadsPF ) | {Name: .Name, PF : .NumReadsPF}&#39;</span>
cat .json <span class="p">|</span> jq <span class="s1">&#39;.Response.Items[] | select(.NumReadsPF ) | &quot;\( .Name)\t\(.NumReadsPF)&quot;&#39;</span>
cat .json <span class="p">|</span> jq <span class="s1">&#39;.Response.Items[] | select(.NumReadsPF &gt; 747912) | &quot;\( .Name)\t\(.NumReadsPF)&quot;&#39;</span>
</pre></div>
</td></tr></table></div>
</div>
<div class="section" id="limitations-of-basemount">
<h2>6. Limitations of BaseMount<a class="headerlink" href="#limitations-of-basemount" title="Permalink to this headline">¶</a></h2>
<p>According to <a class="reference external" href="https://help.basespace.illumina.com/articles/descriptive/introduction-to-basemount/">Illumina</a>,</p>
<p>Every new directory access made by BaseMount relies on FUSE, the BaseSpace API and the user’s credentials. This mechanism means that, as currently available, BaseMount does not support the following types of access:</p>
<ul class="simple">
<li><p>Cluster access, where many compute nodes can access the files. FUSE mounted file systems are per-host and cannot be accessed from many hosts without additional infrastructure.</p></li>
<li><p>BaseMount also doesn’t refresh files or directories. In order to reflect changes done via the Web GUI in your command line tree, you currently need to unmount (basemount –unmount ) and restart BaseMount.</p></li>
<li><p>The Runs Files directory is not mounted automatically for you as there can be 100k + files available in that mount which can take a couple minutes to load for really large runs. You can still mount this directory manually if needed.</p></li>
<li><p>In general, lots of concurrent requests can cause stability issues on a resource constrained system. Keep in mind, this is an early release and stability will increase.</p></li>
</ul>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="mcbl-datadownloads-hudsonalpha.html" class="btn btn-neutral float-right" title="Download from hudsonalpha.org" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="mcbl-share-basespace.html" class="btn btn-neutral float-left" title="Share runs and projects in BaseSpace" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2016-2020, Saranga Wijeratne &amp; Jelmer Poelstra.
      <span class="lastupdated">
        Last updated on Jul 06, 2020.
      </span>

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>